test/linalg.jl

```diff
@@ -0,0 +1,41 @@
+using NiSparseArrays:imul!
+const approx_rtol = 100*eps()
+
+@testset "matrix multiplication" begin
+    for T in (Float64, ComplexF64)
+        for i = 1:5
+            A = sprand(T, 10, 5, 0.5)
+            b = rand(T, 5)
+            outv = A*b
+            c = zero(outv)
+            ioutv = imul!(copy(c), A, b, 1 ,1)[1] # replace with imul!(similar(outv), A, b, 1, 1)[1]?
```

Comment:
Generally, `similar` is faster than `copy` because copy has an extra filling step.

- `similar`: allocate memory
- `copy`: allocate memory + fill the value

Since this is test cases, I wouldn't bother much here.

Reply:
Would similar(c) has the same value with c's value? imul! means c = c+A*b and if we want to evaluate it at A*b, c should be zero value. I tried similar before and it seems to be uninitialized value not zero value?

Reply:
Oh, correct! I forgot that NiLang requires the output to be initialized here. So no, we can't use `similar` to initialize output, it must be initialized as some certain value.